src/rebalancer/bridges/LZUnifiedBridge.sol

```diff
@@ -0,0 +1,264 @@
+    event OftExecutorSet(address indexed underlying, address indexed executor);
+
+    error LZBridge_NotEnoughFees();
+    error LZBridge_ChainNotRegistered();
+    error LZBridge_DestinationMismatch();
+    error LZBridge_DifferentInnerToken();
+    error LZBridge_NoOft();
+    error LZBridge_TokenMismatch();
+    error LZBridge_ExecutorNotSet();
+    error LZBridge_OnlyEndpoint();
+    error LZBridge_BadFrom();
+
+    constructor(address _roles, address _endpoint) BaseBridge(_roles) {
+        endpoint = _endpoint;
+    }
+
+    function setBridgeContract(address underlying, address bridgeContract) external onlyBridgeConfigurator {
+        bridgeContracts[underlying] = bridgeContract;
+        emit BridgeContractSet(underlying, bridgeContract);
+    }
+
+    function setOftExecutorContract(address underlying, address bridgeContract) external onlyBridgeConfigurator {
+        oftExecutors[underlying] = bridgeContract;
+        emit OftExecutorSet(underlying, bridgeContract);
+    }
+
+    // ----------- VIEW ------------
+    /**
+     * @inheritdoc IBridge
+     * @dev use `getOptionsData` for `_bridgeData`
+     */
+    function getFee(uint32 _dstChainId, bytes memory _message, bytes memory)
+        external
+        view
+        returns (uint256)
+    {
+        require(_dstChainId > 0, LZBridge_ChainNotRegistered());
+
+        (MessagingFee memory fees,) = _getFee(_dstChainId, _message);
+        return fees.nativeFee; 
+    }
+
+    // ----------- EXTERNAL ------------
+    /**
+     * @inheritdoc IBridge
+     */
+     function sendMsg(
+        uint256 _extractedAmount,
+        address _market,
+        uint32 _dstChainId,
+        address _token,
+        bytes memory _message,
+        bytes memory
+    )
+        external
+        payable
+        onlyRebalancer
+    {
+        require(_dstChainId > 0, LZBridge_ChainNotRegistered());
+
+        SendMsgLocalVars memory v;
+
+        (v.market,,,) = abi.decode(_message, (address, uint256, uint256, bytes));
+        require(_market == v.market, LZBridge_DestinationMismatch());
+
+        v.underlying = ImTokenMinimal(_market).underlying();
+        if (v.underlying != _token) revert LZBridge_TokenMismatch();
+        if (oftExecutors[v.underlying] == address(0)) revert LZBridge_ExecutorNotSet();
+
+        (v.fees, v.sendParam) = _getFee(_dstChainId, _message);
+        if (msg.value < v.fees.nativeFee) revert LZBridge_NotEnoughFees();
+        require(_extractedAmount == v.sendParam.amountLD, BaseBridge_AmountMismatch());
+
+        v.bridgeContract = bridgeContracts[v.underlying];
+        if (v.bridgeContract == address(0)) {
+            v.bridgeContract = v.underlying;
+        }
+
+        MessagingReceipt memory msgReceipt = _delegateExecuteSend(
+            v.underlying,
+            v.bridgeContract,
+            v.sendParam,
+            v.fees,
+            msg.sender, // rebalancer
+            _market     // refund address
```

Comment:
Refunds are sent to `_market` (dest chain). That’s not the fee payer and may be non-payable, so refunds can be lost or revert. 

Shouldn't refunds should go to the source chain payer, either the rebalancer or the admin?